### PR TITLE
[Mailer] Remove deprecated code

### PR DESCRIPTION
**Pull request by jderusse**
| Q             | A
| ------------- | ---
| Branch?       | 6.0
| Bug fix?      | no
| New feature?  | no
| Deprecations? | no
| Tickets       | -
| License       | MIT
| Doc PR        | -

### CODE REVIEW

File: src/Symfony/Component/Mailer/Bridge/Mailgun/Transport/MailgunApiTransport.php [hunk 1]
```diff
@@ -130,17 +130,7 @@ private function getPayload(Email $email, Envelope $envelope): array
...
             }
 
-            // Check if it is a valid prefix or header name according to Mailgun API
-            $prefix = substr($name, 0, 2);
-            if (\in_array($prefix, ['h:', 't:', 'o:', 'v:']) || \in_array($name, ['recipient-variables', 'template', 'amp-html'])) {
-                $headerName = $name;
-            } else {
-                // fallback to prefix with "h:" to not break BC
-                $headerName = 'h:'.$name;
-                trigger_deprecation('symfony/mailer', '5.1', 'Not prefixing the Mailgun header name with "h:" is deprecated. Use header name "%s" instead.', $headerName);
```
Comment (derrabus) [inline src/Symfony/Component/Mailer/Bridge/Mailgun/Transport/MailgunApiTransport.php]: I wonder if we should keep the check and turn the deprecation into an exception instead.

Comment (jderusse) [reply src/Symfony/Component/Mailer/Bridge/Mailgun/Transport/MailgunApiTransport.php]: Yep, I also hesitate here

Comment (stof) [reply src/Symfony/Component/Mailer/Bridge/Mailgun/Transport/MailgunApiTransport.php]: I strongly think that we should actually *revert* that deprecation as a bugfix (see my review on the PR introducing it, as this deprecation makes it impossible to write code that sends an email with headers, while working with both the mailgun transport and other transport without changes)

**Comment by jderusse** [comment]
Blocked by #41380

**Comment by nicolas-grekas** [comment]
Now unblocked

File: src/Symfony/Component/Mailer/Exception/HttpTransportException.php [hunk 2]
```diff
@@ -20,14 +20,8 @@ class HttpTransportException extends TransportException
 {
     private $response;
 
-    public function __construct(?string $message, ResponseInterface $response, int $code = 0, \Throwable $previous = null)
+    public function __construct(string $message, ResponseInterface $response, int $code = 0, \Throwable $previous = null)
     {
-        if (null === $message) {
-            trigger_deprecation('symfony/mailer', '5.3', 'Passing null as $message to "%s()" is deprecated, pass an empty string instead.', __METHOD__);
```
Comment (derrabus) [inline src/Symfony/Component/Mailer/Exception/HttpTransportException.php:26]: You may remove `symfony/deprecation-contracts` from `symfony/mailer` as well now.

**Review by derrabus — CHANGES_REQUESTED** [review]

File: src/Symfony/Component/Mailer/Exception/HttpTransportException.php (hunk 2 above)
Comment (jderusse) [reply src/Symfony/Component/Mailer/Exception/HttpTransportException.php:26]: good catch. I forgot we added it 4 days ago

**Review by derrabus — APPROVED** [review]

File: src/Symfony/Component/Mailer/Bridge/Amazon/CHANGELOG.md [hunk 3]
```diff
@@ -1,6 +1,12 @@
 CHANGELOG
 =========
 
+6.0
+---
+
+* Remove the `SesApiTransport` class. Use `SesApiAsyncAwsTransport` instead
+* Remove the `SesHttpTransport` class. Use `SesHttpAsyncAwsTransport` instead
```
Comment (nicolas-grekas) [inline src/Symfony/Component/Mailer/Bridge/Amazon/CHANGELOG.md]: missing space at start of line (please rebase while fixing)

Comment (stof) [reply src/Symfony/Component/Mailer/Bridge/Amazon/CHANGELOG.md]: there is no reason to put a space before the `*` of a markdown list

Comment (nicolas-grekas) [reply src/Symfony/Component/Mailer/Bridge/Amazon/CHANGELOG.md]: for markdown no, for consistency yes

**Comment by nicolas-grekas** [comment]
Thank you @jderusse.

**Pull request merged**